routes/campgrounds.js

```diff
@@ -1,8 +1,31 @@
-const   express = require("express"),
-        router = express.Router(),
-        Campground = require("../models/campground"),
-        Comment = require("../models/comments"),
-        middleware = require("../middleware");
+const express = require("express"),
+    router = express.Router(),
+    Campground = require("../models/campground"),
+    Comment = require("../models/comments"),
+    middleware = require("../middleware");
+multer = require('multer');
+storage = multer.diskStorage({
+
+    filename: function (req, file, callback) {
+        callback(null, Date.now() + file.originalname);
+    }
+});
+var imageFilter = function (req, file, cb) {
+    // accept image files only
+    if (!file.originalname.match(/\.(jpg|jpeg|png|gif)$/i)) {
+        return cb(new Error('Only image files are allowed!'), false);
+    }
+    cb(null, true);
+};
+var upload = multer({ storage: storage, fileFilter: imageFilter })
+
+var cloudinary = require('cloudinary');
+cloudinary.config({
```

Comment:
the next time try to make dotenv work. I know you already tried, but this is super important.